### PR TITLE
Dockerfile: update to different 11 jre

### LINKED ISSUE #3045: Fix docker scan vulnerabilities
Checking the [Dockerfile](https://github.com/hyperledger/besu/blob/master/docker/openjdk-11/Dockerfile) seems that base image `openjdk:11.0.7-jre-slim-buster` is bringing some vulnerabilities flagged by some image scanning tools like snyk or harbor.

To get list of vulnerabilities you can run following command:

```
 docker scan -f docker/openjdk-11/Dockerfile hyperledger/besu:develop-openjdk-11 --severity high
```

Producing following output:
```
Tested 91 dependencies for known issues, found 10 issues.

Base Image                  Vulnerabilities  Severity
openjdk:11-jre-slim-buster  62               2 critical, 8 high, 6 medium, 46 low
```

In addition some alternative base images are recommended:

```
Recommendations for base image upgrade:

Alternative image types
Base Image                         Vulnerabilities  Severity
openjdk:17-ea-22-jdk-oracle        0                0 critical, 0 high, 0 medium, 0 low
openjdk:16-ea-33                   0                0 critical, 0 high, 0 medium, 0 low
openjdk:17-ea-10-jdk-oracle        0                0 critical, 0 high, 0 medium, 0 low
openjdk:17-ea-26-jdk-oraclelinux8  0                0 critical, 0 high, 0 medium, 0 low
```

It is not critical but some private docker registry won't allow to pull such images with high severity vulnerabilities.

### PR DESCRIPTION
**Pull request by macfarla**
Signed-off-by: Sally MacFarlane <sally.macfarlane@consensys.net>

Update Dockerfile to use a Java 11 JRE which passes docker scan

## Fixed Issue(s)
Fixes #3045 
## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

**Comment by macfarla** [comment]
Running locally seems to work

```
docker scan -f docker/openjdk-11/Dockerfile hyperledger/besu:21.10.4-SNAPSHOT-openjdk-11 --severity high

Testing hyperledger/besu:21.10.4-SNAPSHOT-openjdk-11...

Organization:      macfarla
Package manager:   deb
Target file:       docker/openjdk-11/Dockerfile
Project name:      docker-image|hyperledger/besu
Docker image:      hyperledger/besu:21.10.4-SNAPSHOT-openjdk-11
Platform:          linux/amd64
Base image:        eclipse-temurin:11-jre
Licenses:          enabled

✓ Tested 144 dependencies for known issues, no vulnerable paths found.
```
and running with `docker run` ->

```
2021-12-13 06:20:52.330+00:00 | EthScheduler-Timer-0 | INFO  | PivotBlockRetriever | Retrieve a pivot block that can be confirmed by at least 5 peers.
2021-12-13 06:20:52.332+00:00 | EthScheduler-Timer-0 | INFO  | PivotBlockConfirmer | Confirm pivot block 13795122 with at least 5 peers.
2021-12-13 06:20:52.568+00:00 | nioEventLoopGroup-3-3 | INFO  | PivotBlockConfirmer | Received 1 confirmation(s) for pivot block header 13795122: 0xfde1e31dfeaea2aa91951c451e26bcd2274243cf62cc21dfe0f6a808e39d19da
2021-12-13 06:20:52.574+00:00 | nioEventLoopGroup-3-10 | INFO  | PivotBlockConfirmer | Received 2 confirmation(s) for pivot block header 13795122: 0xfde1e31dfeaea2aa91951c451e26bcd2274243cf62cc21dfe0f6a808e39d19da
2021-12-13 06:20:52.580+00:00 | nioEventLoopGroup-3-2 | INFO  | PivotBlockConfirmer | Received 3 confirmation(s) for pivot block header 13795122: 0xfde1e31dfeaea2aa91951c451e26bcd2274243cf62cc21dfe0f6a808e39d19da
2021-12-13 06:20:52.633+00:00 | nioEventLoopGroup-3-3 | INFO  | PivotBlockConfirmer | Received 4 confirmation(s) for pivot block header 13795122: 0xfde1e31dfeaea2aa91951c451e26bcd2274243cf62cc21dfe0f6a808e39d19da
2021-12-13 06:20:52.677+00:00 | nioEventLoopGroup-3-1 | INFO  | PivotBlockConfirmer | Confirmed pivot block at 13795122: 0xfde1e31dfeaea2aa91951c451e26bcd2274243cf62cc21dfe0f6a808e39d19da
2021-12-13 06:20:52.688+00:00 | nioEventLoopGroup-3-1 | INFO  | WorldStateDownloader | Begin downloading world state from peers for block 13795122 (0xfde1e31dfeaea2aa91951c451e26bcd2274243cf62cc21dfe0f6a808e39d19da). State root 0x7c65f99dcd70f72af819be3e63253abff4ccffda0e6f6b17fa279b087596e930
2021-12-13 06:20:53.043+00:00 | nioEventLoopGroup-3-3 | INFO  | SyncTargetManager | Found common ancestor with peer Peer 0x65e70d27ac7aed4132... at block 0
```

File: docker/openjdk-11/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 
-FROM openjdk:11-jre-slim-buster
+FROM eclipse-temurin:11-jre
```
Comment (shemnon) [inline docker/openjdk-11/Dockerfile:2]: (a) The variant for this docker file is, explicitly by tag and directory name, `openjdk-11`. If the solutionis to stop using openjdk-11 the the tag and directory name should be changed to `eclipse-temurin-11`

(b) There are two other packages using `openjdk:11-jre-slim-buster` in this repo, any patch should also address those two docker images

**Review by shemnon — DISMISSED** [review]
version changes should either be within naming convention or update the naming convention to match the underlying docker image.

**Review by shemnon — CHANGES_REQUESTED** [review]
version changes should either be within naming convention or update the naming convention to match the underlying docker image.

**Comment by shemnon** [comment]
Alternative proposal: #3171 - change the base linux image.

**Pull request closed without merging**